### DISCUSSION TITLE
How to make custom tokens work?

### DISCUSSION THREAD

**Discussion by d49017** [primefaces/community#1850] [PrimeVue]
I'm trying to add some custom tokens for a custom component, like this:
 ```javascript
 const MyPreset = definePreset(Aura, {
  components: {
    mycomponent: {
      colorScheme: {
        light: {
          color: '#ff0000'
        }
      }
    }
  }
})
 ```
 but unfortunately when I check the :root{} in browser there is no `--<prefix>-mycomponent-color` css variable. How can I make it work?

(This is a crucial functionality for us, to be able to define our own tokens that will be transformed into the css variables, as the ones that are present are not enough).

**Comment by johnatawnclementawn** [comment]
It's been my understanding that tokens defined in `components` are only applied to Primevue components. To expose tokens to my custom components, I have added them in the `semantic` tokens group. This exposes my tokens as you detailed above `--<prefix>-mycomponent-color`, but it feels counterintuitive because I'm defining a token that's being used for a specific component (e.g. `mycomponent`), not at the semantic level like primary or surface...
